pkg/operator/ceph/object/objectstore.go

```diff
@@ -56,129 +62,448 @@ type idType struct {
+	}
+	logger.Debugf("got zone ID for zone %v", objContext.Zone)
+
+	if zoneID == zoneGroupJson.MasterZoneID {
+		logger.Debugf("zone is master")
+		return true, nil
+	}
+
+	logger.Debugf("zone is not master")
+	return false, nil
+}
+
+func checkZoneGroupIsMaster(objContext *Context) (bool, error) {
+	logger.Debugf("checking if zone group %v is the master zone group", objContext.ZoneGroup)
+	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+
+	zoneGroupOutput, err := RunAdminCommandNoMultisite(objContext, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+
+	zoneGroupJson, err := DecodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse master zone id")
+	}
+
+	zoneGroupIsMaster, err := strconv.ParseBool(zoneGroupJson.IsMaster)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse is_master from zone group json into bool")
+	}
+
+	return zoneGroupIsMaster, nil
+}
+
+func DecodeSecret(secret *v1.Secret, keyName string) (string, error) {
+	realmKey, ok := secret.Data[keyName]
+
+	if !ok {
+		return "", errors.New("key was not in secret data")
+	}
+
+	return string(realmKey), nil
+}
+
+func GetRealmKeyArgs(context *clusterd.Context, realmName, namespace string) (string, string, error) {
+	logger.Debugf("getting keys for realm %v", realmName)
+	// get realm's access and secret keys
+	realmSecretName := realmName + "-keys"
+	realmSecret, err := context.Clientset.CoreV1().Secrets(namespace).Get(realmSecretName, metav1.GetOptions{})
+	if err != nil {
+		return "", "", errors.Wrapf(err, "failed to get realm %q keys secret", realmName)
+	}
+	logger.Debugf("found keys secret for realm %v", realmName)
+
+	accessKey, err := DecodeSecret(realmSecret, AccessKeyName)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "failed to decode realm %q access key", realmName)
+	}
+	secretKey, err := DecodeSecret(realmSecret, SecretKeyName)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "failed to decode realm %q access key", realmName)
+	}
+	logger.Debugf("decoded keys for realm %v", realmName)
+
+	accessKeyArg := fmt.Sprintf("--access-key=%s", accessKey)
+	secretKeyArg := fmt.Sprintf("--secret-key=%s", secretKey)
+
+	return accessKeyArg, secretKeyArg, nil
+}
+
+func getZoneEndpoints(objContext *Context, serviceEndpoint string) ([]string, error) {
+	logger.Debugf("getting current endpoints for zone %v", objContext.Zone)
+	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+
+	zoneGroupOutput, err := RunAdminCommandNoMultisite(objContext, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return []string{}, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zoneGroupJson, err := DecodeZoneGroupConfig(zoneGroupOutput)
```

Comment:
It's already been decoded, right? no need for json in the name.
```suggestion
	zoneGroups, err := DecodeZoneGroupConfig(zoneGroupOutput)
```